it/src/test/java/org/corfudb/universe/scenario/AddNodeDuringInvalidateLayoutIT.java

```diff
@@ -0,0 +1,157 @@
+                client.getRuntime().invalidateLayout();
+                return invoke.callRealMethod();
+            }).when(baseClientSpy).reset();
+
+            // When we try to issue a reset, depending on when the router was pruned,
+            // either a TimeoutException will be thrown
+            // (because the connection future times out),
+            // or a NetworkException will be thrown
+            // (because the connection future completed exceptionally).
+            assertThatThrownBy(() -> CFUtils.getUninterruptibly(baseClientSpy.reset(),
+                    TimeoutException.class, NetworkException.class))
+                    .isInstanceOfAny(NetworkException.class, TimeoutException.class);
+
+            // Now lets create a client router that is not part of a router pool,
+            // issue a reset to the node we would want to add.
+            // When a reset is called, the layout will be invalidated.
+            try (NettyClientRouter clientRouter = new NettyClientRouter(NodeLocator.parseString(endpoint),
+                    client.getRuntime().getParameters())) {
+
+                baseClient = new BaseClient(clientRouter, layout.getEpoch(), layout.getClusterId());
+                BaseClient spyBaseClient = Mockito.spy(baseClient);
+
+                Mockito.doAnswer(invoke -> {
+                    client.getRuntime().invalidateLayout();
+                    return invoke.callRealMethod();
+                }).when(spyBaseClient).reset();
+
+                // The reset should go through.
+                boolean resetHappened = spyBaseClient.reset()
+                        .get(waitForResetDurationSeconds, TimeUnit.SECONDS);
+                assertThat(resetHappened).isTrue();
+            }
+
+            // Wait until a node fully restarts.
+            Sleep.sleepUninterruptibly(Duration.ofMillis(resetWaitPeriod));
+
+            // Now lets create a client router that is not part of a router pool,
+            // and run the add node workflow.
+            // Every time a clientRouter sends a request, the layout will be invalidated.
+            try (NettyClientRouter clientRouter = new NettyClientRouter(NodeLocator.parseString(endpoint),
+                    client.getRuntime().getParameters())) {
+
+                NettyClientRouter spyRouter = Mockito.spy(clientRouter);
+
+                spyRouter.addClient(new LayoutHandler())
+                        .addClient(new ManagementHandler())
+                        .addClient(new BaseHandler());
+
+                Mockito.doAnswer(invoke -> {
+                    client.getRuntime().invalidateLayout();
+                    return invoke.callRealMethod();
+                }).when(spyRouter).sendMessageAndGetCompletable(Mockito.any(), Mockito.any());
+
+                // Run add node.
+                client.getManagementView()
+                        .addNode(endpoint, numRetries, timeOutDuration, pollDuration);
```

Comment:
I fixed this line here in my branch, right now here it does don't really tests the invalidation during any part of the addNode.